vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -550,11 +551,28 @@ void AdsImpl::SetConfirmationsIsReady(
   is_confirmations_ready_ = is_ready;
 }
 
-std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory() {
+std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory(
+  const Ads::AdHistoryFilterType adHistoryFilterType) {
   std::map<uint64_t, std::vector<AdsHistory>> ads_history;
   base::Time now = base::Time::Now().LocalMidnight();
 
   auto ad_history_details = client_->GetAdsShownHistory();
+
+  std::unique_ptr<AdHistoryFilter> adHistoryFilter = nullptr;
+  switch (adHistoryFilterType) {
+    case (Ads::AdHistoryFilterType::CONFIRMATION_TYPE_TRUMPING_POLICY): {
+      adHistoryFilter = std::make_unique<AdHistoryConfirmationFilter>();
```

Comment:
Seems like a factory would be helpful here, but I guess there's also no need to over-engineer it until we get to that point.

Reply:
Yep, agreed - same for the Exclusion/Permission Rules. When it's appropriate we can get to start thinking factories :)